sdk/python/kfp/containers/_component_builder.py

```diff
@@ -229,6 +275,8 @@ def build_python_component(
     base_image = default_base_image_or_builder
     if isinstance(base_image, Callable):
       base_image = base_image()
+  if not dependency:
+    dependency = []
```

Comment:
nit: Just curious, do you see any benefit of changing the argument default from `None` to `[]` and then do a check here?

Reply:
In general we should prevent using empty list (or anything mutable) as default, because of the way Python evaluate default values.

See for example, https://nikos7am.com/posts/mutable-default-arguments/#:~:text=Avoid%20using%20an%20empty%20list%20as%20a%20default%20argument%20to%20a%20function,-June%204%2C%202018&text=This%20is%20not%20the%20behavior,time%20the%20function%20is%20called.